middle_end/flambda/inlining/inlining_decision.mli

```diff
@@ -32,9 +32,12 @@ module Function_declaration_decision : sig
   val can_inline : t -> bool
 end
 
+(* If a cost_metrics is passed it will be used as the cost metrics for
+   the function declaration instead of grabbing the cost metrics from the
+   one stored on the code located in the env. *)
 val make_decision_for_function_declaration
    : Downwards_env.t
-  -> ?params_and_body:Function_params_and_body.t
+  -> ?cost_metrics:Cost_metrics.t
```

Comment:
I think it would be better if this parameter were not optional.  Maybe declare a new type:
```
type cost_metrics_source = From_denv | Metrics of Cost_metrics.t
```